crates/compiler/src/obj_gen/mod.rs

```diff
@@ -238,10 +231,64 @@ impl ObjectGenerator {
             }
         }
 
+        // Finally, we generate initializers for globals that have them.
+        for global in module.get_globals() {
+            if global.get_initializer().is_some() {
+                let global_name = global.get_name().to_str()?;
+                let Some(global_id) = data.map.module_globals.get_by_left(global_name) else {
```

Comment:
I'm getting increasingly nervous about the bidict impl we're using for globals. Now that we're using them so heavily, having a true bidirectional mapping sounds more and more risky -- I'm growing worried we'll at some point have aliasing/two symbols pointing to the same address, and then we'll wind up with one symbol missing without realizing it.

Thoughts on handling that soon?

Reply:
I do not ever call `get_by_right`, so I have turned this into a standard map instead.